pkg/db/db.go

```diff
@@ -0,0 +1,107 @@
+package db
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+	"go.uber.org/zap"
+)
+
+type DB struct {
+	Host      string `yaml:"host"`
+	Port      int    `yaml:"port"`
+	User      string `yaml:"user"`
+	Password  string `yaml:"password"`
+	Name      string `yaml:"name"`
+	TableName string `yaml:"providers_table_name"`
+}
+
+func getConnectionString(db *DB) string {
+	return fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", db.Host, db.Port, db.User, db.Password, db.Name)
+}
+
+type Postgres struct {
+	db   *DB
+	pool *pgxpool.Pool
+}
+
+func NewPGInstance(ctx context.Context, logger *zap.Logger, db *DB) (*Postgres, error) {
```

Comment:
Why do we need to explicitly specify that this is a PG instance?
```suggestion
func NewClient(ctx context.Context, logger *zap.Logger, db *DB) (*Postgres, error) {
```